### DISCUSSION TITLE
If the rate limiting is set on the worker, would a new deployment reset duration in rate limiting?

### DISCUSSION THREAD

**Discussion by BoKKeR** [taskforcesh/bullmq#2582] [Q&A] [answered]
We are setting the rate limiting on the worker. Lets take the following example: 

```const worker = new Worker('painter', async job => paintCar(job), {
  limiter: {
    max: 10,
    duration: 1 * 1000,
  },

```

Assuming from this each 1 second, the worker would consumer 10 jobs.Adding a extra worker, now with 2 workers we should still only take 10 jobs total from the queue every 1 second. Is this right? Or would each worker take 10 jobs/s

Given that. If we have a long duration. Lets say for limiting an API for free tier each month. 

```const worker = new Worker('painter', async job => paintCar(job), {
  limiter: {
    max: 10,
    duration:  31 * 24 * 60 * 1000,
  },
```

Would this result in only 10 jobs being processed (in a small burst in the beginning of the month), even if we increase the amount of deployments? Or scale down deployments to 0 for and back to 1? Or would the duration reset.

There is some misunderstanding on my side as in where is the worker code is running. Is the worker present on redis? Is it present only within the node application?

**Comment by BoKKeR** [comment]
To answer my own question it works. 

I can set 1 month time and 10 limit, this processes 10 requests each month, with maximum processing 10! 

The problem others might run into is that there is an upper limit to setTimeout value which will throw warnings once you hit a 30 day mark! 

```
node:19) TimeoutOverflowWarning: 2511078580 does not fit into a 32-bit signed integer.                       
Timeout duration was set to 1.
```

**Reply by roggervalf** [reply]
hi @BoKKeR. sorry for the delay and thanks for letting us know, this should be addressed soon

**Reply by roggervalf** [reply, marked as answer]
Pls upgrade to v5.12.2

**Reply by BoKKeR** [reply]
Thanks, that solved the issue
